core/module.h

```diff
@@ -155,6 +158,12 @@ class ModuleBuilder {
 
 class ModuleTask;
 
+enum CheckResults {
```

Comment:
Could you give this a more descriptive name (since I could imagine other things being checked) and add a comment?  For the name, another option is to move it someplace more general and leave it as is.